### PR TITLE
Put test output in build directory

### PR DESCRIPTION
**Pull request by jdahm**
## Technical Description

Put GTClang integration test output in the build directory.

### CODE REVIEW

File: dawn/src/dawn/Compiler/DawnCompiler.cpp [hunk 1]
```diff
@@ -88,6 +88,16 @@ struct ComputeEditDistance<std::string> {
   }
 };
 
+std::pair<std::string, std::string> dirnameBasename(const std::string& filename) {
+
+  const char sep = '/';
+
+  const auto i = filename.rfind(sep);
```
Comment (havogt) [inline dawn/src/dawn/Compiler/DawnCompiler.cpp]: I saw a lot of path manipulation in prs recently. We should consider using std::filesystem.

Comment (jdahm) [reply dawn/src/dawn/Compiler/DawnCompiler.cpp]: Ah, yes I’m glad you said that. There are different parts of the code all using their own parsing methods. I will change that in this PR before we merge.

File: gtclang/test/utils/gtclang-tester/gtclang_tester/runner.py [hunk 2]
```diff
@@ -123,14 +131,19 @@ def run(self):
             # Check expected files
             if test.get_expected_file() and not Config.generate_reference:
                 for files in test.get_expected_file():
-                    for outputfile, referencefile in zip(files.get_output_files(), files.get_reference_files()):
+                    for outputfile, referencefile in zip(
+                        files.get_output_files(), files.get_reference_files()
+                    ):
 
                         try:
+                            print("HERE ", outputfile, referencefile)
```
Comment (jdahm) [inline gtclang/test/utils/gtclang-tester/gtclang_tester/runner.py]: Delete this line before merging!

File: dawn/src/dawn/Compiler/DawnCompiler.cpp (hunk 1 above)
Comment (jdahm) [reply dawn/src/dawn/Compiler/DawnCompiler.cpp]: I used it in this place. We'll have to track down the other places later.

**Comment by jdahm** [comment]
I would like to merge this at the same time as #598.

**Comment by jdahm** [comment]
launch jenkins

**Comment by jdahm** [comment]
@havogt This works locally but is breaking on Travis. Can you attempt a quick fix tomorrow? It looks like it's missing the c++ filesystem library in the link step. From a quick search online I found out that this may be due to a difference in clang vs. gcc and cmake.

**Comment by jdahm** [comment]
Closed in favor of #598.

**Pull request closed without merging**